src/H5system.c

```diff
@@ -979,7 +979,7 @@ char *
 Wstrcasestr_wrap(const char *haystack, const char *needle)
 {
     if (needle && !*needle)
-        return haystack;
+        return (char *)haystack;
```

Comment:
We're kind of cheating here, but this is what everyone else does when they implement this behavior.